Fix Fury Warrior back-to-back Whirlwind recommendations

joshjau opened the pull request:
Issue:
- Fixes issue where Whirlwind was recommended occasionally back-to-back in single target
- Adds proper buff duration check before reapplication
- Maintains correct stack count (4 with Meat Cleaver, 2 with Improved Whirlwind)
- Improves usability check to prevent unnecessary Whirlwind casts

Changes:
1. Modified handler to check buff duration before reapplication
2. Kept existing usability check for single target scenarios
3. Maintained proper stack application logic

joshjau committed:
Fix Fury Warrior back-to-back Whirlwind recommendations

Issue:
- Fixes issue where Whirlwind was recommended occasionally back-to-back in single target
- Adds proper buff duration check before reapplication
- Maintains correct stack count (4 with Meat Cleaver, 2 with Improved Whirlwind)
- Improves usability check to prevent unnecessary Whirlwind casts

Changes:
1. Modified handler to check buff duration before reapplication
2. Kept existing usability check for single target scenarios
3. Maintained proper stack application logic

syrifgit commented on TheWarWithin/WarriorFury.lua:
```diff
@@ -1872,12 +1872,16 @@ spec:RegisterAbilities( {
 
         usable = function ()
             if action.taunt.known and action.heroic_throw.known and settings.check_ww_range and not ( action.taunt.in_range and not action.heroic_throw.in_range ) then return false, "target is outside of whirlwind range" end
+            if active_enemies == 1 and buff.meat_cleaver.up then return false, "meat cleaver already active" end
```
I feel like this should just be handled by the APL. Can you create a snapshot pre-change where the recommendation is wrong?

syrifgit commented on TheWarWithin/WarriorFury.lua:
```diff
@@ -1872,12 +1872,16 @@ spec:RegisterAbilities( {
-            if talent.improved_whirlwind.enabled then
-                applyBuff( "meat_cleaver", nil, talent.meat_cleaver.enabled and 4 or 2 )
+            if talent.improved_whirlwind.enabled or talent.meat_cleaver.enabled then
+                -- Only apply if buff is down or about to expire
```
Does it not refresh the buff? If it had 2 seconds remaining would another cast not refresh it to 4?

syrifgit reviewed (commented):
Couple questions

joshjau commented:
Thanks for the review @syrifgit!

While the APL handles the basic talent check, these changes add important optimizations:

1. The usability check `if active_enemies == 1 and buff.meat_cleaver.up then return false` prevents unnecessary rage spending by blocking redundant Whirlwind casts in single-target when the buff is already active.

2. The range check has been simplified to be more direct and maintainable, removing the complex proxy checks through taunt and heroic throw ranges.

These optimizations work alongside the APL to prevent inefficient ability usage while maintaining the core rotation logic.

joshjau commented:
updated comment.

joshjau committed:
Improve Whirlwind buff handling with readyTime

Per Syrif discussion.

Changes:
- Switches from usable to readyTime for better buff prediction
- Returns buff.meat_cleaver.remains instead of false for single target
- Maintains existing range check functionality

This improves Whirlwind recommendations by letting the APL handle buff refresh timing while preventing unnecessary casts.

joshjau commented:
updated pr pull after discussion with Syrif.

joshjau commented:
i will close this for now, given a few others are open that should be more successful to merge and will continue to investigate this with Syrif.
